test/context_flush_test.rb

```diff
@@ -366,6 +366,7 @@ def test_tracer_hard_limit_overrides_soft_limit
     tracer.configure(min_spans_before_partial_flush: context.max_length,
                      max_spans_before_partial_flush: context.max_length,
                      partial_flush_timeout: 3600)
+    tracer.writer = FauxWriter.new
```

Comment:
Nasty side effect of this is that everytime we reconfigure the tracer, it automatically overrides the previous writer with a new one, which puts test tracers in a bad state.

Reply:
It might be good to add a comment here about that

Reply:
Unfortunately this comment will apply just about everywhere we use `tracer.configure` in tests so I don't think it'll help. I'd rather try to find some way to get rid of this, because this kind of practice in tests is not sustainable.

Reply:
Something `dd-trace-py` did for tests in create a `DummyTracer` to use in all tests that will always overwrite the writer to be a `DummyWriter`, even after calling `.configure()`

Reply:
Yeah, I was looking at that, too. Problem is `configure` applies a bunch of mutations to `Writer`, so if you have to re-initialize with a `FauxWriter`, then you have to copy over all the settings, otherwise you'll end up with a misconfigured `Writer`.

I think ideally, the code should either construct immutable `Writer` objects that have to be replaced, or mutable ones that are never replaced; right now we're in a kind of unhappy middle ground.

Nonetheless I'm going to try to avoid such refactoring right now, maybe there's something else I can do to work around this in the mean time.

Reply:
yeah, we have that same issue. since we isolated it into one place in a subclass of `Tracer` it wasn't too bad... enough to get by for now

Reply:
For sure, I'll try one of these tricks to make something work for now, but it would be nice if we refactored things to be more testable. A goal for the future, perhaps.